File: mybatis/src/main/java/data/spring/mybatis/application/required/ProductRepository.java
package data.spring.mybatis.application.required;

import data.spring.mybatis.application.service.command.ProductSearchCommand;
import data.spring.mybatis.application.service.command.ProductUpdateCommand;
import data.spring.mybatis.domain.Product;

import java.util.List;
import java.util.Optional;

public interface ProductRepository {
    void save(Product productEntity);

    int saveAll(List<Product> productEntities);

    void update(ProductUpdateCommand updateCommand);

    Optional<Product> findById(Long productId);

    List<Product> findAll(ProductSearchCommand searchCommand);

    int deleteAll();
}

File: src/main/kotlin/data/spring/mybatis/application/required/product/ProductRepository.kt
package data.spring.mybatis.application.required.product

import data.spring.mybatis.application.service.product.command.ProductSearchCommand
import data.spring.mybatis.domain.product.Product

interface ProductRepository {
    fun save(product: Product): Int

    fun truncate(): Int

    fun findById(productId: Long): Product?
    fun findWithCond(searchCommand: ProductSearchCommand): List<Product>
}